examples/air-quality-sensor-app/linux/AirQualitySensorAppAttrUpdateDelegate.cpp

```diff
@@ -0,0 +1,314 @@
+    { "RadonConcentrationMeasurement",
+      { &RadonConcentrationMeasurement::Attributes::MeasuredValue::Set,
+        &RadonConcentrationMeasurement::Attributes::MinMeasuredValue::Get,
+        &RadonConcentrationMeasurement::Attributes::MinMeasuredValue::Set,
+        &RadonConcentrationMeasurement::Attributes::MaxMeasuredValue::Get,
+        &RadonConcentrationMeasurement::Attributes::MaxMeasuredValue::Set } }
+};
+
+AirQualitySensorAttrUpdateHandler * AirQualitySensorAttrUpdateHandler::FromJSON(const char * json)
+{
+    Json::Reader reader;
+    Json::Value value;
+
+    if (!reader.parse(json, value))
+    {
+        ChipLogError(NotSpecified,
+                     "AllClusters App: Error parsing JSON with error %s:", reader.getFormattedErrorMessages().c_str());
+        return nullptr;
+    }
+
+    if (value.empty() || !value.isObject())
+    {
+        ChipLogError(NotSpecified, "AllClusters App: Invalid JSON command received");
+        return nullptr;
+    }
+
+    if (!value.isMember("Name") || !value["Name"].isString())
+    {
+        ChipLogError(NotSpecified, "AllClusters App: Invalid JSON command received: command name is missing");
+        return nullptr;
+    }
+
+    return Platform::New<AirQualitySensorAttrUpdateHandler>(std::move(value));
+}
+
+void AirQualitySensorAttrUpdateHandler::HandleCommand(intptr_t context)
+{
+    auto * self             = reinterpret_cast<AirQualitySensorAttrUpdateHandler *>(context);
+    std::string clusterName = self->mJsonValue["Name"].asString();
+
+    VerifyOrReturn(!self->mJsonValue.empty(), ChipLogError(NotSpecified, "Invalid JSON event command received"));
+
+    if (clusterName == "AirQuality")
+    {
+        uint8_t newValue = static_cast<uint8_t>(self->mJsonValue["NewValue"].asUInt());
+        self->OnAirQualityChangeHandler(newValue);
+    }
+    else if (clusterName == "TemperatureMeasurement")
+    {
+        int16_t newValue = static_cast<int16_t>(self->mJsonValue["NewValue"].asUInt());
+        self->OnTemperatureChangeHandler(newValue);
+    }
+    else if (clusterName == "RelativeHumidityMeasurement")
+    {
+        uint16_t newValue = static_cast<uint16_t>(self->mJsonValue["NewValue"].asUInt());
+        self->OnHumidityChangeHandler(newValue);
+    }
+    else if (clusterName.find("Concentration") != std::string::npos)
+    {
+        float newValue = static_cast<uint16_t>(self->mJsonValue["NewValue"].asFloat());
+        self->OnConcetratorChangeHandler(clusterName, newValue);
+    }
+    else
+    {
+        ChipLogError(NotSpecified, "Invalid cluster name %s", clusterName.c_str());
+    }
+
+    ChipLogError(NotSpecified, "done");
+}
+
+void AirQualitySensorAttrUpdateHandler::OnAirQualityChangeHandler(uint8_t newValue)
+{
+    EndpointId endpoint  = 1;
+    EmberAfStatus status = AirQuality::Attributes::AirQuality::Set(endpoint, static_cast<AirQuality::AirQualityEnum>(newValue));
```

Comment:
Why is the cluster not exposing APIs for this?  Followup, please...

Reply:
@bzbarsky-apple We thought it was ok to use the generated accessors from `Accessors.h`?

Reply:
Ok, thanks - I'll raise some issues on our side and we'll see if we can get to it before v1.2